### PR TITLE
Add dependency and trait diagrams to markdown reporting

### LINKED ISSUE #499: Generate inheritance diagram for contracts and traits within a project
For instance, the diagram can be produced in the [mermaid](https://mermaid.js.org) format and added to markdown docs on GitHub (for Tact as well). This is useful for auditing purposes as well: to quickly grasp the overall structure of the project.

The diagram should also show the standard traits (all the way down).

### PR DESCRIPTION
**Pull request by Gusarich**
Closes #499 

- [x] I have updated CHANGELOG.md
- [ ] ~~I have documented my contribution in Tact Docs: https://github.com/tact-lang/tact-docs/pull/PR-NUMBER~~
- [ ] ~~I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases~~
- [x] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

**Comment by Gusarich** [comment]
@novusnota @anton-trunov will this work as you want?

upd: I just realized I didn't handle the whole dependency tree, but anyway what do you think about this place for diagrams?

**Comment by anton-trunov** [comment]
@Gusarich It actually might! Thanks! Could you post the mermaid output for some contract example with traits? GitHub should be able to render it here in comments

**Comment by Gusarich** [comment]
@anton-trunov here's an example for SampleJetton project from Tact repo:

# the result for Jetton Minter:
## Trait Inheritance Diagram

```mermaid
graph TD
SampleJetton --> BaseTrait
SampleJetton --> Jetton
Jetton --> BaseTrait
Jetton --> Ownable
Ownable --> BaseTrait
SampleJetton --> Ownable
Ownable --> BaseTrait
```

## Contract Dependency Diagram

```mermaid
graph TD
SampleJetton --> JettonDefaultWallet
```

# the result for Jetton Wallet:
## Trait Inheritance Diagram

```mermaid
graph TD
JettonDefaultWallet --> BaseTrait
```

## Contract Dependency Diagram

```mermaid
graph TD
```

**Comment by anton-trunov** [comment]
Looks really cool!

There are a couple of issues that we need to address:

- Why `Ownable` inherits two times from `BaseTrait`?
- for Jetton Wallet the contract inheritance diagram is empty

**Comment by Gusarich** [comment]
@anton-trunov fixed!
Now the diagram looks this way:

# Trait Inheritance Diagram

```mermaid
graph TD
SampleJetton
SampleJetton --> BaseTrait
SampleJetton --> Jetton
Jetton --> BaseTrait
Jetton --> Ownable
Ownable --> BaseTrait
SampleJetton --> Ownable
```

# Contract Dependency Diagram

```mermaid
graph TD
SampleJetton
SampleJetton --> JettonDefaultWallet
```


# Trait Inheritance Diagram

```mermaid
graph TD
JettonDefaultWallet
JettonDefaultWallet --> BaseTrait
```

# Contract Dependency Diagram

```mermaid
graph TD
JettonDefaultWallet
```

**Comment by anton-trunov** [comment]
Wow! this looks so good!

@novusnota we should add that to tact-docs!

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - `-e` / `--eval` CLI flags to evaluate constant Tact expressions: PR [#462](https://github.com/tact-lang/tact/pull/462)
 - `-q` / `--quiet` CLI flags to suppress compiler log output: PR [#509](https://github.com/tact-lang/tact/pull/509)
+- Markdown report for compiled contracts now include Mermaid diagrams for trait inheritance and contract dependencies: PR [#560](https://github.com/tact-lang/tact/pull/560)
```
Comment (anton-trunov) [inline CHANGELOG.md]:
```suggestion
- Markdown report for compiled contracts now includes Mermaid diagrams for trait inheritance and contract dependencies: PR [#560](https://github.com/tact-lang/tact/pull/560)
```

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**